indico/web/client/js/react/components/dates/CalendarSingleDatePicker.jsx

```diff
@@ -5,47 +5,41 @@
 // modify it under the terms of the MIT License; see the
 // LICENSE file for more details.
 
-import 'react-dates/initialize';
 import PropTypes from 'prop-types';
-import React from 'react';
-import {DayPickerSingleDateController as DayPicker} from 'react-dates';
+import React, {useRef} from 'react';
 
-import {renderMonthElement, responsiveReactDates} from './util';
+import {DatePickerGrid, DatePickerInlineCalendar} from 'indico/react/components/DatePickerCalendar';
+import {useNativeEvent} from 'indico/react/hooks';
 
-import 'react-dates/lib/css/_datepicker.css';
 import '../style/dates.scss';
 
-export default class CalendarSingleDatePicker extends React.Component {
-  static propTypes = {
-    disabledDate: PropTypes.func,
-    yearsBefore: PropTypes.number,
-    yearsAfter: PropTypes.number,
-  };
-
-  static defaultProps = {
-    disabledDate: () => false,
-    yearsBefore: 5,
-    yearsAfter: 5,
-  };
-
-  state = {
-    focused: true,
-  };
-
-  onFocusChange = ({focused}) => {
-    this.setState({focused});
-  };
-
-  render() {
-    const {focused} = this.state;
-    const {disabledDate, yearsBefore, yearsAfter, ...props} = this.props;
-    return responsiveReactDates(DayPicker, {
-      ...props,
-      focused,
-      onFocusChange: this.onFocusChange,
-      isOutsideRange: disabledDate,
-      hideKeyboardShortcutsPanel: true,
-      renderMonthElement: params => renderMonthElement(yearsBefore, yearsAfter, params),
-    });
-  }
+export default function CalendarSingleDatePicker({onChange, date, minDate, maxDate, ...props}) {
```

Comment:
This one has a bug when the second date you click is before the first one.
The "field-based" date range picker adjusts the start date when you do that, but this one updates the end date so you're in a weird state where nothing is displayed, and when you click apply you end up with a reverse range where the end date is before the start date.

In case this is confusing (it sounds confusing to me :D):

- Go to `/rooms/book?recurrence=daily&number=1&interval=week&sd=2025-01-20&ed=2025-01-24&st=13%3A00&et=14%3A00`
- Click on "Mon 20 Jan - Fri 24 Jan"
- Click on Feb 14
- Click on Feb 11
- (click "Apply")

Also interesting: If I click on Feb 11 a second time, it correctly sets it as the start date and I end up in end date editing mode.

Reply:
I gotcha. I'll fix that on Monday.

Reply:
ty, works fine now